iocore/net/quic/test/test_QUICHandshakeProtocol.cc

```diff
@@ -102,16 +102,32 @@ TEST_CASE("QUICHandshakeProtocol")
   BIO *key_bio(BIO_new_mem_buf(server_key, sizeof(server_key)));
   EVP_PKEY *pkey = PEM_read_bio_PrivateKey(key_bio, nullptr, nullptr, nullptr);
   SSL_CTX_use_PrivateKey(server_ssl_ctx, pkey);
+  SSL_CTX_set_alpn_select_cb(
+    server_ssl_ctx,
+    [](SSL *ssl, const unsigned char **out, unsigned char *outlen, const unsigned char *in, unsigned inlen, void *) {
+      auto ret = SSL_select_next_proto(const_cast<unsigned char **>(out), outlen,
+                                       reinterpret_cast<const unsigned char *>("\6h3-foo"), 6, in, inlen);
```

Comment:
The openssl docs aren't clear on this, but I believe the server_len parameter here should be 7 instead of 6. The prefix of '\6' in the server string is correct, but I think that the server_len has to count all characters in the server string, inclusive of any prefixes.

I think I'm reading the code right:
https://github.com/openssl/openssl/blob/3f71add9e57fb48cb5efdc765860daf754db40e9/ssl/ssl_lib.c#L2896

It will need to loop over all characters in server and client.